pkg/client/restclient/config.go

```diff
@@ -50,6 +50,8 @@ type Config struct {
 	// be appended to all request URIs used to access the apiserver. This allows a frontend
 	// proxy to easily relocate all of the apiserver endpoints.
 	Host string
+	// UnixSocket is a path to the unix socket that the (proxy) apiserver is listening on.
+	UnixSocket string
```

Comment:
@lavalamp This "http+unix" scheme is something specific to this library (which isn't even written in Go), I don't think that's a scheme which will work out of the box. If I'm wrong, I'd appreciate a proof that it's working with the default http transport in Go.

@krousey Good idea, I will try it. I thought about returning an error when both variables are being tried to set, but your proposition sounds better.